app/frontend/helpers/validators.ts

```diff
@@ -73,33 +68,48 @@ const donationAmountValidator = (fieldValue, coins, balance) => {
   return null
 }
 
-const feeValidator = (sendAmount, transactionFee, donationAmount, balance) => {
+const txPlanValidator = (sendAmount, balance, txPlan, donationAmount = 0) => {
+  const transactionFee = txPlan.fee || txPlan.estimatedFee
+
   if (transactionFee >= balance) {
     return {code: 'SendAmountCantSendAnyFunds'}
   }
-  if (donationAmount > 0 && sendAmount + transactionFee + donationAmount > balance) {
+  if (sendAmount + transactionFee > balance) {
     return {
-      code: 'DonationInsufficientBalance',
+      code: 'SendAmountInsufficientFunds',
       params: {balance},
     }
   }
-  if (sendAmount + transactionFee > balance) {
+  if (donationAmount > 0 && sendAmount + transactionFee + donationAmount > balance) {
     return {
-      code: 'SendAmountInsufficientFunds',
+      code: 'DonationInsufficientBalance',
       params: {balance},
     }
   }
+  if (txPlan.error) return txPlan.error
   return null
 }
 
-const delegationFeeValidator = (fee, deposit, balance) => {
-  if (fee + deposit > balance) {
+const delegationPlanValidator = (balance, txPlan) => {
+  const transactionFee = txPlan.fee || txPlan.estimatedFee
+  const deposit = txPlan.deposit || 0
+  if (transactionFee + deposit > balance) {
     return {
-      code: 'DelegationAccountBalanceError',
+      code: 'DelegationBalanceError',
       params: {balance},
     }
   }
-  return null
+  const txPlanError = txPlanValidator(0, balance, txPlan)
+  return txPlanError || null
+}
+
+const withdrawalPlanValidator = (sendAmount, balance, txPlan) => {
```

Comment:
I would rename sendAmount to rewardsAmount, since that's basically what being sent to the function and it fits more.

Reply:
Fixed